File: exapmles.proj/xcode.proj/autolayout/trymasonry/trymasonry/ViewController.m
//
//  ViewController.m
//  trymasonry
//
//  Created by CZ on 5/3/16.
//  Copyright © 2016 projm. All rights reserved.
//

#import <Masonry/Masonry.h>

#import "ViewController.h"

@interface ViewController ()
@property (nonatomic, strong) UIView *topLeft;
@property (nonatomic, strong) UIView *topRight;
@property (nonatomic, strong) UIView *bottom;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    _topLeft = [UIView new];
    [_topLeft setBackgroundColor: [UIColor yellowColor]];
    _topRight = [UIView new];
    [_topRight setBackgroundColor:[UIColor redColor]];
    _bottom = [UIView new];
    [_bottom setBackgroundColor:[UIColor blueColor]];
    [self.view addSubview:_topLeft];
    [self.view addSubview:_topRight];
    [self.view addSubview:_bottom];
    
    
    // AutoLayout with Masonry
    [_topLeft mas_makeConstraints:^(MASConstraintMaker *make) {
        make.width.equalTo(_topRight.mas_width);
        make.top.equalTo(self.view.mas_top).offset(20);
        make.left.equalTo(self.view.mas_left).offset(10);
        make.right.equalTo(_topRight.mas_left).offset(-10);
        make.bottom.equalTo(_bottom.mas_top).offset(-10);
    }];

    [_topRight mas_makeConstraints:^(MASConstraintMaker *make) {
        //make.width.equalTo(_topLeft);
        make.top.equalTo(self.view.mas_top).offset(20);
        make.right.equalTo(self.view.mas_right).offset(-10);
        //make.left.equalTo(_topLeft.mas_left).offset(10);
        make.bottom.equalTo(_bottom.mas_top).offset(-10);
    }];
    
    [_bottom mas_makeConstraints:^(MASConstraintMaker *make) {
        make.height.equalTo(_topLeft.mas_height);
        make.left.equalTo(self.view.mas_left).offset(10);
        make.right.equalTo(self.view.mas_right).offset(-10);
        make.bottom.equalTo(self.view.mas_bottom).offset(-10);
    }];
    

}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: blog/iOS/ui/uilabel/example/UILabelDemo/UILabelDemo/ViewController.swift
//
//  ViewController.swift
//  UILabelDemo
//
//  Created by apollo on 15/11/2016.
//  Copyright © 2016 projm. All rights reserved.
//

import UIKit
import CoreText

class ViewController: UIViewController {

    func layoutLabel () {
        let lbl = UILabel()
        lbl.text = "I'm a sample label"

        
        
        //lbl.adjustsFontSizeToFitWidth = true
        //lbl.lineBreakMode = .byWordWrapping
        //lbl.allowsDefaultTighteningForTruncation = true

        lbl.baselineAdjustment = .alignBaselines

//        lbl.shadowColor = UIColor.red
//        lbl.shadowOffset = CGSize(width: 0, height: 5)
//        lbl.textRect(forBounds: lbl.frame, limitedToNumberOfLines: 0)
        
        lbl.backgroundColor = UIColor.yellow
        lbl.frame = CGRect(x: 10, y: 48, width: 200, height: 60)
        let txt = "I'm a attribute Text"
        let attrText = NSMutableAttributedString(string: txt )
        attrText.addAttribute(NSUnderlineStyleAttributeName, value: 2, range: NSRange(location: 0, length: txt.lengthOfBytes(using: .utf8)))
        lbl.attributedText = attrText
        self.view.addSubview(lbl)
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        self.layoutLabel()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

